lib/FacebookIdModule.js

```diff
@@ -17,22 +18,98 @@ const https = require('https');
  *  TODO: Update the README with FB's Contract
  */
 class FacebookIdModule {
-
     /**
-     * TODO: Implement the below method
      * HTTP request to identity provider with supplied token.
      * If [principalId = 0] is returned, the token is invalid or the call otherwise failed.
      * @param token
      * @returns principalId
      */
     callIdProvider(token) {
+
         var principalId = 0;
-        return principalId;
+        var input_token='input_token=' + token;
+        var access_token = '&access_token='
+            .concat(identityProvider.FACEBOOK.appId)
+            .concat('|')
+            .concat(identityProvider.FACEBOOK.appSecret);
+
+        var tokenedPath = identityProvider.FACEBOOK.path.concat(input_token).concat(access_token);
+        var disposableClone = {};
+        Object.assign(disposableClone, identityProvider.FACEBOOK);
+        disposableClone.path = tokenedPath;
+        return new Promise(
+            /*executor*/
+            function (resolve, reject) {
+                var req = https.get(disposableClone, (response) => {
+                    var str = '';
+
+                    response.on('data', (chunk) => {
+                        str += chunk;
+                    });
+
+                    response.on('error', () => {
+                        resolve(principalId);
+                    });
+
+                    response.on('end', () => {
+                        logger.info('No more data in response.');
+
+                        var json = JSON.parse(str);
+
+                        logger.info(json);
+
+                        if (json.data && json.data.is_valid && json.data.user_id) {
+                            principalId = json.data.user_id;
+                        }
+                        resolve(principalId);
+                    });
+                });
+                req.end();
+
+            });
+
     }
 }
 
 var identityProvider = {
-    FACEBOOK: {}
+    FACEBOOK: {
+        host: "graph.facebook.com",
+        path:"/debug_token?", //from what I can tell this is OK for production use
+        port: 443,
+        appId:'1219028494814465',
+        appSecret:'8fb31f4e7ce440be94f25de797568fe2'
+    }
 }
 
-module.exports = FacebookIdModule;
```

Comment:
This is required to use the module in the `AuthorizerModule.js`. Running `npm run build` shows how the tests break with this change